.github/workflows/e2e_aws.yaml

```diff
@@ -0,0 +1,173 @@
+
+      - name: Use oras to get qcow2 from ${{ inputs.podvm_image }}
+        if: ${{ inputs.oras }}
+        run: |
+          oras pull ${{ inputs.podvm_image }}
+          tar xvJpf podvm.tar.xz
+          qcow2=$(find ./*.qcow2)
+          echo "PODVM_QCOW2=$(pwd)/${qcow2}" >> "$GITHUB_ENV"
+        working-directory: src/cloud-api-adaptor/podvm
+
+      - name: Install kustomize
+        run: |
+          command -v kustomize >/dev/null || \
+          curl -s "https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh" | \
+            sudo bash -s /usr/local/bin
+
+      - name: Update kustomization configuration
+        run: |
+          cd "install/overlays/aws"
+          kustomize edit set image "cloud-api-adaptor=${{ inputs.caa_image }}"
+          # Print for debugging
+          echo "::group::aws kustomization"
+          cat kustomization.yaml
+          echo "::endgroup::"
+
+      - name: Config aws
+        run: |
+          echo "CAA_IMAGE=\"${{ inputs.caa_image }}\"" >> aws.properties
+          echo "disablecvm=\"true\"" >> aws.properties
+          echo "cluster_type=\"${{ inputs.cluster_type }}\"" >> aws.properties
+          #[ "${{ inputs.cluster_type }}" = "onprem" ] && echo "use_public_ip=\"true\"" >> aws.properties
+          echo "ssh_kp_name=\"caa-e2e-test\"" >> aws.properties
+          # For debugging
+          echo "::group::aws.properties"
+          cat aws.properties
+          echo "::endgroup::"
+
+        # Note: aws cli is already installed on github's runner image.
+      - name: Config aws CLI
+        run: |
+          aws configure set aws_access_key_id ${{ secrets.AWS_ACCESS_KEY_ID }} --profile default
+          aws configure set aws_secret_access_key ${{ secrets.AWS_SECRET_ACCESS_KEY }} --profile default
+          aws configure set region us-east-1 --profile default
+
+      - name: Create on-prem cluster
+        if: inputs.cluster_type == 'onprem'
+        run: |
+          # Let's use kcli to build a kubeadm cluster for us
+          echo "::group::Configure libvirt"
+          ./libvirt/config_libvirt.sh
+          # Add the kcli install directory to PATH for later steps
+          echo "${HOME}/.local/bin" >> "$GITHUB_PATH"
+          echo "::endgroup::"
+          export CONTAINER_RUNTIME=${{ inputs.container_runtime }}
+          echo "::group::Create cluster with $CONTAINER_RUNTIME"
+          ./libvirt/kcli_cluster.sh create
+          echo "KUBECONFIG=$HOME/.kcli/clusters/peer-pods/auth/kubeconfig" >> "$GITHUB_ENV"
+          echo "::endgroup::"
+
+      - name: run tests
+        id: runTests
+        run: |
+          export CLOUD_PROVIDER=aws
+          export DEPLOY_KBS=false
+          export TEST_PROVISION="yes"
+          export TEST_TEARDOWN="yes"
+          export TEST_PROVISION_FILE="$PWD/aws.properties"
+          export TEST_PODVM_IMAGE="${{ env.PODVM_QCOW2 }}"
+          export TEST_E2E_TIMEOUT="90m"
+
+          make test-e2e
```

Comment:
Can we get any debug logs in case things go wrong?

Reply:
Good idea, I overlooked it completely. In a next PR hopefully